Tests/SentryTests/Performance/SentryTracerTests.swift

```diff
@@ -235,6 +235,28 @@ class SentryTracerTests: XCTestCase {
         assertAppStartMeasurementNotPutOnTransaction()
     }
     
+    func testFinish_WithUnfinishedChildren() {
+        let sut = fixture.getSut(waitForChildren: false)
+        let child1 = sut.startChild(operation: fixture.transactionOperation)
+        let child2 = sut.startChild(operation: fixture.transactionOperation)
+        let child3 = sut.startChild(operation: fixture.transactionOperation)
+        child2.finish()
+        fixture.currentDateProvider.setDate(date: Date(timeIntervalSinceNow: 0.1))
+        sut.finish()
```

Comment:
`m`: I think we should test with the `DefaultCurrentDateProvider` here because this test doesn't fail when removing `span.timestamp = self.timestamp;` in `SentryTracer.m`.
```suggestion
        CurrentDate.setCurrentDateProvider(DefaultCurrentDateProvider.sharedInstance())
        let sut = fixture.getSut(waitForChildren: false)
        let child1 = sut.startChild(operation: fixture.transactionOperation)
        let child2 = sut.startChild(operation: fixture.transactionOperation)
        let child3 = sut.startChild(operation: fixture.transactionOperation)
        child2.finish()
        sut.finish()
```